Simple menu structure

ksaliya opened the pull request:
Hey, I've added a gameState variable to your code and added couple of buttons using which one can start and restart the game. Please consider pulling. Thanx a lot.

ksaliya committed:
Added a simple game menu

ksaliya committed:
Modified to use the game menu

strikemike2k reviewed (changes requested):
Excellent work. I like how much cleaner the draw loop is with each part being split into its own function.
First, we need to include the `p5.dom.js` file in the `index.html` for those `createButton()` methods and `removeElements()` to work. Second, though not required, maybe it would be a good idea to stop the draw loop on the `initGame()` and `endGame()` functions by adding `noLoop();` to the bottom of those functions. We can then restart the draw loop after the user presses the button by adding `loop();` to the bottom of the `startGame()` function. I just don't see a need to continue the loop while the menu screen is up because it will keep creating buttons one after another.

ksaliya committed:
included the p5.dom.js library

ksaliya committed:
included Loop() and noLoop()

looping was disabled in initGame() and endGame() while the looping was enabled only when the game is running.

ksaliya committed:
Code cosmetics

removed some commented lines

ksaliya commented:
Hey mike, Thanks for the response and feedback. You are absolutely correct, it makes no sense to keep the loop running when there are no changes being made. So, I included noLoop() and loop() functions.

strikemike2k reviewed (approved):
Great work. Perfect!
